phase2/kubeadm/configure-vm-kubeadm-master.sh

```diff
@@ -48,62 +51,100 @@ networking:
   podSubnet: "$POD_NETWORK_CIDR"
 EOF
 
-# handle v1alpha1
-########################################################################
-if [[ "$KUBEADM_KUBERNETES_VERSION_MINOR" -lt "11" ]]; then
+  # add cloud provider
+  if [[ "$KUBEADM_ENABLE_CLOUD_PROVIDER" == true ]]; then
+    cat <<EOF |tee -a $KUBEADM_CONFIG_FILE
+apiServerExtraArgs:
+  cloud-provider: "$CLOUD_PROVIDER"
+EOF
+  fi
 
-  # add api version and token
-  cat <<EOF |tee -a $KUBEADM_CONFIG_FILE
+  # set ipvs
+  if [[ "$KUBEPROXY_MODE" == "ipvs" ]]; then
+    cat <<EOF |tee -a $KUBEADM_CONFIG_FILE
+kubeProxy:
+  config:
+    featureGates:
+      SupportIPVSProxyMode: true
+    mode: "$KUBEPROXY_MODE"
+EOF
+  fi
+
+  # add feature gates
+  KUBEADM_FEATURE_GATES=`echo "$KUBEADM_FEATURE_GATES" | sed -e 's/^[[:space:]]*//'`
+  if [[ ! -z $KUBEADM_FEATURE_GATES ]]; then
+    foptions=`echo $KUBEADM_FEATURE_GATES | sed -e 's/=/: /g;s/,/\\\n  /g'`
+    cat <<EOF |tee -a $KUBEADM_CONFIG_FILE
+featureGates:
+  `echo -e "$foptions"`
+EOF
+  fi
+
+  # handle v1alpha1 or v1alpha2
+  if [[ "$KUBEADM_KUBERNETES_VERSION_MINOR" -lt "11" ]]; then
+    cat <<EOF |tee -a $KUBEADM_CONFIG_FILE
 apiVersion: kubeadm.k8s.io/v1alpha1
 token: "$KUBEADM_TOKEN"
 EOF
+  else
+    cat <<EOF |tee -a $KUBEADM_CONFIG_FILE
+apiVersion: kubeadm.k8s.io/v1alpha2
```

Comment:
Again?

Reply:
in an earlier PR that did a similar change here, it was discussed whether we should use `migrate` vs using bash to write different config versions.

the problem here is that, say the old v1alpha1 config would not pass with kubeadm and it needs `migrate`.
migrate on the other hand is unreliable because we remove versions at the beginning of the cycle which breaks master, thus it was decided to handle the config flavors in bash.

i don't have plans to touch the alpha1 and alpha2 configs anymore once we merge this.
only add beta1 this cycle and hopefully nothing else as we are moving to cluster-api testing. :)